pkg/reconciler/v1alpha1/route/route.go

```diff
@@ -413,8 +423,57 @@ func objectRef(a accessor, gvk schema.GroupVersionKind) corev1.ObjectReference {
 	}
 }
 
-func routeDomain(ctx context.Context, route *v1alpha1.Route) string {
+// Save these as global to save time on each call to routeDomain()
+var savedTemplate *template.Template = nil
+var savedText string = ""
+
+func routeDomain(ctx context.Context, route *v1alpha1.Route) (string, error) {
 	domainConfig := config.FromContext(ctx).Domain
 	domain := domainConfig.LookupDomainForLabels(route.ObjectMeta.Labels)
-	return fmt.Sprintf("%s.%s.%s", route.Name, route.Namespace, domain)
+
+	// These are the available properties they can choose from.
+	// We could add more over time - e.g. RevisionName if we thought that
+	// might be of interest to people.
+	data := struct {
+		Name      string
+		Namespace string
+		Domain    string
+	}{
+		Name:      route.Name,
+		Namespace: route.Namespace,
+		Domain:    domain,
+	}
+
+	networkConfig := config.FromContext(ctx).Network
+	text := networkConfig.RouteTemplate
+
+	// Either this is the first time or the user-provided RouteTemplate changed
+	if savedTemplate == nil || text != savedText {
```

Comment:
This caching behavior should be part of the config flow. It needs to become `DynamicConfig`. The current implementation is potentially racy.

Reply:
I agree that it's racy but I wasn't sure if we supported 2 controllers or not - I can fix that.
However, can you elaborate on what you mean by "part of the config flow"?
Do I need to use DynamicConfig if it's read-only in this func? Why doesn't "domainConfig" need it?

Reply:
Yes, presumably it will never surface as a real data race (however our controllers have a threadiness of 2 IIRC, so I might be wrong as well).

It is however a (to me) unexpected way of caching values. You can have a look at the autoscaler and how DynamicConfig is used there. Essentially, I'd like us to cache configuration related values in the configuration itself. Could we add the parsed template to the configuration?